### ISSUE TITLE
Tube noise scaling

### ISSUE THREAD

**Issue by msyriac** [simonsobs/mapsims#43]
Currently on `master` in `SONoiseSimulator`, the noise curves are obtained at each frequency for the default number of tubes. As I understand it, this noise curve from so_noise_models corresponds to what would be obtained if all the tube arrays at that particular frequency were coadded. (@mhasself ?) I think this leads to an inconsistency when hitmap version v0.2 are used, wherein one requests a dichroic noise map for each tube. In this case, the co-added noise curve is always used, thus leading to a tube noise map that is less noisy than the forecast code requires it to be.

**Comment by msyriac** [comment]
If this is indeed the case, I propose that the noise curves be calculated on the fly (in the simulate function) so as to adapt to requests that are based on either "channels" or "tubes". I've checked that the actual noise curve call takes a negligible amount of compute time. Since the hitmap will already have been loaded (and cached) when `simulate` starts, this also has the advantage that we can switch to calls to the noise curve code with fsky defined in the more natural way that @damonge suggested in https://github.com/simonsobs/mapsims/pull/5#discussion_r244939311 , based on <Nhits>, rather than with fsky=1, which makes the code a bit harder to parse.

**Comment by zonca** [comment]
I think you are right, good job finding this out!
So I think this affects all of the LAT maps in [`202002_noise`](https://github.com/simonsobs/map_based_simulations/tree/master/202002_noise) except LT6. It doesn't affect SAT maps.

```
tubes = {
    "LT0": ["UHF1", "UHF2"],
    "LT1": ["UHF1", "UHF2"],
    "LT2": ["MFF1", "MFF2"],
    "LT3": ["MFF1", "MFF2"],
    "LT4": ["MFS1", "MFS2"],
    "LT5": ["MFS1", "MFS2"],
    "LT6": ["LF1", "LF2"],
    "ST0": ["UHF1", "UHF2"],
    "ST1": ["MFF1", "MFF2"],
    "ST2": ["MFS1", "MFS2"],
    "ST3": ["LF1", "LF2"],
}
```

Hitmaps V2 always assume you are working with tubes (see docstring of `MapSim`), so what about loading the spectra for half the focal plane for LAT UHF and MF (only if we use hitmaps v2)?

in `noise.py`:

```
            survey = getattr(so_models, self.LA_noise_model)(                                                                        
                sensitivity_mode=self.sensitivity_mode,                                                                              
                survey_efficiency=self.survey_efficiency,                                                                            
                survey_years=self.LA_years,                                                                                          
                N_tubes=[self.LA_number_LF, self.LA_number_MF, self.LA_number_UHF],                                                  
                el=self.elevation,                                                                                                   
            )
```

I would add:

```
N_tubes=[self.LA_number_LF, self.LA_number_MF, self.LA_number_UHF],                                                  

# with hitmaps v0.2, we simulate LAT tubes independently, therefore MF and UHF bands
# are simulated twice, so we scale the number of tubes accordingly here
if self.hitmap_version = "v0.2":
    N_tubes[1] /= 2
    N_tubes[2] /= 2
```

**Comment by msyriac** [comment]
I like the general direction of your suggestion but I'm still confused. Shouldn't it say
```
if self.hitmap_version = "v0.2":
    N_tubes = [1,1,1]
```

Also,  the current default for number of tubes for SAT is [1,4,2]. Should it be [1,2,1] for v0.1? (and [1,1,1] again for v0.2).

**Comment by zonca** [comment]
not sure, I find it confusing.

Docstring from `so_noise_models` says:

```
          N_tubes: A list of tuples giving the survey-averaged number
            of each LAT tube in operation.  For example, the default
            is [('LF', 1), ('MF', 4), ('UHF', 2)], populating a total
            of 7 tubes in this LAT.  Fractional tubes are acceptable
            (imagine a tube were swapped out part way through the
            survey).
```

Let's focus on LAT first, see:
https://github.com/simonsobs/so_noise_models/blob/master/so_models_v3/SO_Noise_Calculator_Public_v3_1_1.py#L462

that `1 4 2` is default

so if we split those into 2 maps each, we should have `1 2 1`, but to be more general I divide by 2.

**Comment by msyriac** [comment]
Ok, for the LAT v0.2, each simulation requests one of the tubes LT0 - LT6. For each of these requests made with my suggestion of [1,1,1], the number of tubes from the noise curve code that actually end up counting are:
```
LT0 [0 0 1] (UHF)
LT1 [0 0 1] (UHF)
LT2 [0 1 0] (MF)
LT3 [0 1 0] (MF)
LT4 [0 1 0] (MF)
LT5 [0 1 0] (MF)
LT6 [1 0 0] (LF)
```
where I've zeroed out the tube-types that aren't used because they don't have frequencies from the requested tube. Summing up those vectors gives you [1,4,2], the original that matches v0.1. I think this suggests we should be calling the noise curve with [1,1,1] for v0.2. Do you agree?

**Comment by msyriac** [comment]
(I think the 2 maps / dichroic aspect is a red herring.)

**Comment by msyriac** [comment]
For the SAT, I think the following should be done.

First:
![2020-04-24_16:58:51](https://user-images.githubusercontent.com/3674397/80256612-01431a00-864d-11ea-9f2f-aba52d6b069f.png)
so the default numbers in the class init should be:
```
SA_number_LF=0.4
SA_number_MF=1.6
SA_number_UHF=1
```
which should be used for v0.1.

For v0.2, it's a bit more complicated. The following suggestion should work if implemented:
```
N_tubes = [0.4,0,0] if ST3
N_tubes = [0,0,1] if ST0
N_tubes = [0,1,0] if ST1 *
N_tubes = [0,0.6,0] if ST2 *
```
* I'm not sure whether ST1 and ST2 should be swapped -- the one with the higher hit counts should get [0,1,0]


```

**Comment by zonca** [comment]
ok, thank you for the clarification, yes, everything makes sense now, I'll check the hitcounts of ST1 and ST2.
Then I can implement this.

**Comment by msyriac** [comment]
I can do the implementation in my new PR since I've been mulling it over for a while now. (Do let me know what you find out about ST1 vs. ST2 though) In my implementation, I want to make a bit clearer that there are two "modes" you can generate sims in:

### channel or v0.1 mode

 - Combinations like LAT-145 or SAT-93 are requested                                                                                   
 - used if you just want the co-added performance at a given frequency                                                                
 - you are limited to v0.1 of the hitmaps                                                                                    
 - you can still use full_covariance, but you have to specify a pair of channels                                                      
 - hitmaps for all frequencies are identical   
                                                                                                       
### tube or v0.2 mode 

 - Combinations like LT0 or ST1 are requested                                                                                   
 - you get back a map pair from a given tube                                                                                              
 - hitmaps (from v0.2) are specific to each tube map pair                                                                                       
 - you can turn on or off full_covariance  

Do you agree this general dichotomy would be nice, in principle?

**Comment by zonca** [comment]
I was wondering if we can directly remove support for 0.1 mode, that would simplify the code a lot.

**Comment by msyriac** [comment]
I would be ok with that as well, it would definitely simplify things a lot! People running early debug codes might like to have co-added maps at a given frequency though, but what one can do for that is to request e.g. LT4 with the noise scaled to a new white noise value that matches what the co-added level is. (scaling the noise level is a feature I want to add anyway)

Another possibility is to add an option in the noise simulator to request a simulation of what happens when the map-maker maps e.g. LT2,LT3,LT4,LT5 together -- this can be achieved simply by changing the N_tubes parameter. So the implementation is easy, we just have to think about what the API will look like for that

Given all this, I will go ahead and simplify my new PR to only work with "tubes".

**Comment by zonca** [comment]
agreed! thanks

**Comment by zonca** [comment]
checked the hitmaps, in MSS-0001, ST1 and ST2 have the same total hitcount. actually they are exactly the same.

**Comment by zonca** [comment]
ok, I added a notice at https://github.com/simonsobs/map_based_simulations/tree/master/202002_noise and notified the `pwg-mbs` slack channel about this.

**Comment by msyriac** [comment]
Resolved with #44
